src/katgpucbf/utils.py

```diff
@@ -53,3 +58,60 @@ def parse_source(value: str) -> list[tuple[str, int]] | str:
         return [(ep.host, ep.port) for ep in endpoints]
     except ValueError:
         return value
+
+
+class DeviceStatus(Enum):
+    """Discrete `device-status` readings.
+
+    Doesn't follow the convention of :class:`aiokatcp.Sensor.Status` because
+    it's useful for `UNKNOWN` to have the highest (or worst) value.
+    """
+
+    OK = 1
+    DEGRADED = 2
+    FAIL = 3
+
+
+class DeviceStatusSensor(aiokatcp.AggregateSensor[DeviceStatus]):
+    """Summary sensor for quickly ascertaining device status.
+
+    This takes its value from the worst status of its target set of sensors, so
+    it's quick to identify if there's something wrong, or if everything is good.
+    """
+
+    def __init__(self, target: aiokatcp.SensorSet) -> None:
+        super().__init__(
+            target=target, sensor_type=DeviceStatus, name="device-status", description="Overall engine health"
+        )
+
+    def update_aggregate(
+        self,
+        updated_sensor: aiokatcp.Sensor[_T] | None,
+        reading: aiokatcp.Reading[_T] | None,
+        old_reading: aiokatcp.Reading[_T] | None,
+    ) -> aiokatcp.Reading[DeviceStatus] | None:  # noqa: D102
+        if reading is not None and old_reading is not None and reading.status == old_reading.status:
+            return None  # Sensor didn't change state, so no change in overall device status
+        # Otherwise, we have to recalculate.
+        worst_status: aiokatcp.Sensor.Status = aiokatcp.Sensor.Status.NOMINAL
```

Comment:
Is it necessary to explicitly specify the type? I would think mypy would know the type based on the assigned value.

Reply:
I might have started off with this not being obvious. I can remove it.